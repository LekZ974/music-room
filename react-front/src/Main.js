import React from 'react';

import { connect } from 'react-redux';
import { Font, AppLoading } from 'expo';
import { Ionicons } from '@expo/vector-icons';
import { lifecycle } from 'recompose';
import Routes from './routes';

const withStatus = lifecycle({
  async componentDidMount() {
    /* eslint-disable */
    try {
      await Font.loadAsync({
        Roboto: require('native-base/Fonts/Roboto.ttf'),
        Roboto_medium: require('native-base/Fonts/Roboto_medium.ttf'),
        ...Ionicons.font,
      });
      this.setState({ ready: true });
    } catch (error) {
      console.log(error)
    }
    /* eslint-enable */
  },
});

const Main = withStatus((status, props) => {
  const { isLogged } = props;

  return status.ready ? <Routes screenProps={{ isLogged }} /> : <AppLoading />;
});

const mapStateToProps = state => {
  return {
    isLogged: !!state.user.email,
  };
};

export default connect(
  mapStateToProps,
  null,
)(Main);
