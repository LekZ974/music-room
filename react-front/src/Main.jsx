import React from 'react';
import { StyleProvider } from 'native-base';
import { Text } from 'react-native';

import getTheme from './native-base-theme/components';
import variables from './native-base-theme/variables/commonColor';

const Main = () => (
  <StyleProvider style={getTheme(variables)}>
    <Text>MUSIC ROOM!!!</Text>
  </StyleProvider>
);
export default Main();
