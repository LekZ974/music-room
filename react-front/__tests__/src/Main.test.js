import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Main from '../../src/Main';

jest.mock('react-native-gesture-handler', () => {});
jest.mock('../../src/i18n', () => {});
jest.mock('../../src/routes.js', () => {});

describe('Main snapshot', () => {
  it('renders the loading screen', async () => {
    let wrapper = shallow(<Main />)
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});