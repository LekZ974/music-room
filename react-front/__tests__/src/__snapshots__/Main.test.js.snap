// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshot renders the loading screen 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
